Remove non-standard `Range.compareNode()`

Ahmad-S792 opened the pull request:
#### 3183fdb64fb2abafc508f458ba691dbebaf567c9
<pre>
Remove non-standard `Range.compareNode()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=249114">https://bugs.webkit.org/show_bug.cgi?id=249114</a>
<a href="https://rdar.apple.com/103499504">rdar://103499504</a>

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch removes non-standard `compareNode`, which is not in web specification [1]
and it was removed from Blink in 2015 while from Firefox in version 1.9 (long ago).

[1] <a href="https://dom.spec.whatwg.org/#interface-range">https://dom.spec.whatwg.org/#interface-range</a>

Based on analysis by Karl, there are only 84 instances of this on GitHub, which
links to all archived or deprecated projects (e.g., two of them are WYSIWYG editors).

Additionally, since it is already removed from other browsers safely, this has very little
Website Compatibility risk.

* Source/WebCore/dom/Range.cpp:
(WebCore::Range::compareNode const): Deleted.
* Source/WebCore/dom/Range.h:
* Source/WebCore/dom/Range.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMRange.cpp:
(webkit_dom_range_compare_node):
* Source/WebKitLegacy/mac/DOM/DOMRange.h:
* Source/WebKitLegacy/mac/DOM/DOMRange.mm:
(-[DOMRange compareNode:]):
* LayoutTests/fast/dom/Range/ranges-across-trees.html: Rebaselined
* LayoutTests/fast/dom/Range/ranges-across-trees-expected.txt: Rebaselined
* LayoutTests/fast/dom/Range/range-compareNode-expected.txt: Removed.
* LayoutTests/fast/dom/Range/range-compareNode.html: Removed.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3183fdb64fb2abafc508f458ba691dbebaf567c9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83401 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3018 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/37689 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/88477 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34411 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/85489 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3103 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10973 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64880 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22625 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75789 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45164 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2170 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29992 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/33460 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73272 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/30720 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89853 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/10665 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7692 "Found 1 new test failure: fast/selectors/selection-window-inactive-stroke-color.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73312 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10890 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/71611 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/72538 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16757 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15490 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2001 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/10620 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16092 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10471 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13942 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12243 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMRange.cpp:
```diff
@@ -486,19 +486,7 @@ WebKitDOMDocumentFragment* webkit_dom_range_create_contextual_fragment(WebKitDOM
-        return 0;
-    }
-    return result.releaseReturnValue();
+    return NULL;
```
Why are we returning NULL? We should return 0 instead.

Ahmad-S792 replied:
Sure, will update. :-)

Ahmad-S792 replied:
Done in [3183fdb](https://github.com/WebKit/WebKit/commit/3183fdb64fb2abafc508f458ba691dbebaf567c9).

rniwa commented on Source/WebKitLegacy/mac/DOM/DOMRange.mm:
```diff
@@ -236,10 +236,7 @@ - (DOMDocumentFragment *)createContextualFragment:(NSString *)html
-    if (!refNode)
-        raiseTypeErrorException();
-    return raiseOnDOMError(IMPL->compareNode(*core(refNode)));
+    return NULL;
```
Ditto. This should return 0, not NULL.

Ahmad-S792 replied:
Sure, will update. :-)

Ahmad-S792 replied:
Done in [3183fdb](https://github.com/WebKit/WebKit/commit/3183fdb64fb2abafc508f458ba691dbebaf567c9).

Ahmad-S792 committed:
Remove non-standard `Range.compareNode()`

https://bugs.webkit.org/show_bug.cgi?id=249114
rdar://103499504

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch removes non-standard `compareNode`, which is not in web specification [1]
and it was removed from Blink in 2015 while from Firefox in version 1.9 (long ago).

[1] https://dom.spec.whatwg.org/#interface-range

Based on analysis by Karl, there are only 84 instances of this on GitHub, which
links to all archived or deprecated projects (e.g., two of them are WYSIWYG editors).

Additionally, since it is already removed from other browsers safely, this has very little
Website Compatibility risk.

* Source/WebCore/dom/Range.cpp:
(WebCore::Range::compareNode const): Deleted.
* Source/WebCore/dom/Range.h:
* Source/WebCore/dom/Range.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMRange.cpp:
(webkit_dom_range_compare_node):
* Source/WebKitLegacy/mac/DOM/DOMRange.h:
* Source/WebKitLegacy/mac/DOM/DOMRange.mm:
(-[DOMRange compareNode:]):
* LayoutTests/fast/dom/Range/ranges-across-trees.html: Rebaselined
* LayoutTests/fast/dom/Range/ranges-across-trees-expected.txt: Rebaselined
* LayoutTests/fast/dom/Range/range-compareNode-expected.txt: Removed.
* LayoutTests/fast/dom/Range/range-compareNode.html: Removed.

darinadler commented on LayoutTests/fast/dom/Range/ranges-across-trees.html:
```diff
@@ -38,7 +38,7 @@
         shouldThrowErrorName("bodyRange().comparePoint(shadowNode, 0)", "WrongDocumentError");
         shouldBeFalse("bodyRange().intersectsNode(shadowNode)");
-        shouldThrowErrorName("bodyRange().compareNode(shadowNode)", "WrongDocumentError");
+        shouldThrowErrorName("bodyRange().compareNode(shadowNode)", "TypeError");
```
I don't think it's helpful to keep this. We have the WPT test to be sure it's no longer implemented. We don't need a wrong document test for a function that does not exist.

darinadler reviewed (approved):
There is real risk here of app incompatibility with either calls to the Objective-C method or embedded JavaScript. This is something that other web browsers don't have to concern themselves with, so we can't say it's safe because of them.
